pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -473,35 +497,54 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*operationexecutor.Vo
 	return volumeToMount, nil
 }
 
-func (rc *reconciler) updateStates(volumeToMount *operationexecutor.VolumeToMount) error {
-	err := rc.actualStateOfWorld.MarkVolumeAsAttached(
-		volumeToMount.VolumeName, volumeToMount.VolumeSpec, "", volumeToMount.DevicePath)
-	if err != nil {
-		return fmt.Errorf("Could not add volume information to actual state of world: %v", err)
-	}
-	err = rc.actualStateOfWorld.AddPodToVolume(
-		volumeToMount.PodName,
-		types.UID(volumeToMount.PodName),
-		volumeToMount.VolumeName,
-		nil,
-		volumeToMount.OuterVolumeSpecName,
-		volumeToMount.DevicePath)
-	if err != nil {
-		return fmt.Errorf("Could not add pod to volume information to actual state of world: %v", err)
+func (rc *reconciler) updateStates(volumesNeedUpdate map[api.UniqueVolumeName]*operationexecutor.VolumeToMount) error {
+	// Get the node status to retrieve volume device path information.
```

Comment:
I think since we can get this information out of node state, then just put it. Otherwise devicePath will be empty and DeviceOpened will not work.